pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,165 @@
+// Package baremetal collects bare metal specific configuration.
+package baremetal
+
+import (
+	"fmt"
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	baremetaldefaults "github.com/openshift/installer/pkg/types/baremetal/defaults"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// Platform collects bare metal specific configuration.
+func Platform() (*baremetal.Platform, error) {
+	var libvirtURI, ironicURI, externalBridge, provisioningBridge, apiVIP, ingressVIP string
+	var hosts []*baremetal.Host
+
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Libvirt Connection URI",
+				Help:    "The libvirt connection URI to be used.",
+				Default: baremetaldefaults.LibvirtURI,
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &libvirtURI)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Ironic Connection URI",
+				Help:    "The ironic connection URI to be used.",
+				Default: baremetaldefaults.IronicURI,
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
+		},
+	}, &libvirtURI)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &externalBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
+		},
+	}, &provisioningBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "API VIP",
+				Help:    "The VIP to be used for internal API communication. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.APIVIP,
+			},
+			Validate: apiVIPValidator,
+		},
+	}, &apiVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Ingress VIP",
+				Help:    "The VIP to be used for ingress. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.IngressVIP,
+			},
+			Validate: ingressVIPValidator,
+		},
+	}, &ingressVIP)
+	if err != nil {
+		return nil, err
+	}
+
+	// Keep prompting for hosts
```

Comment:
the installer doesn't ask users for number of replicas for control plane and compute.
for most platforms this is defaulted to 3, 3.
what's the default configuration for bare metal?
and I think it would be best if we prompt for that many only.

Reply:
Tracked in https://github.com/openshift-metal3/kni-installer/issues/101

Reply:
I suggested in another thread that we just drop TUI support completely for now ...